### ISSUE TITLE
firebase_ml_vision docs should recommend setting deployment target to 9.0

### ISSUE THREAD

**Issue by pulyaevskiy** [flutter/flutter#22791]
I needed latest version of `firebase_ml_vision` because version on Pub depends on very old versions of Firebase pods and I'm depending on some Firestore functionality introduced recently.

Looking at latest master I found that someone already removed version constraints from the plugin, so I tried to use it, but it's failing with an error.

## Steps to Reproduce

1. Using macOS: Run `flutter create` to create a new app
2. Add `firebase_ml_vision` to dependencies and point to use version from master branch

```yaml
dependencies:
  firebase_ml_vision:
    git:
      url: git@github.com:flutter/plugins.git
      path: packages/firebase_ml_vision
```
3. Run `flutter run` (assuming you have iOS Simulator open)

## Logs

```
Launching lib/main.dart on iPhone XS Max in debug mode...
Starting Xcode build...
 ├─Assembling Flutter resources...                    1.3s

 └─Compiling, linking and signing...                  3.4s

Xcode build done.                                            9.1s
Failed to build iOS app
Error output from Xcode build:
↳
    2018-10-07 18:39:48.466 xcodebuild[33673:41649681]  DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/DVTFrameworks/DVTFrameworks-14317.18/DVTFoundation/Utilities/DVTFileSystemUtilities.mm:142
    Details:  lstat('/Users/anatoly/Projects/tmp/mk_vision/build/ios/Debug-iphonesimulator/GoogleMobileVision/GoogleMVTextDetectorResources.bundle') failed with errno 2 (No such file or directory)
    Function: DVTRecursiveMkdirResult dvt_recursiveMkdir_returningErrorString(NSString *__strong, NSString *__autoreleasing *)
    Thread:   <NSThread: 0x7f88e7601180>{number = 3, name = (null)}
    Please file a bug at https://bugreport.apple.com with this warning message and any useful information you can provide.
    ** BUILD FAILED **

Xcode's output:
↳
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Debug ===
    ld: targeted OS version does not support use of thread local variables in l1862 for architecture x86_64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
Could not build the application for the simulator.
Error launching application on iPhone XS Max.
```

Flutter doctor

```
[✓] Flutter (Channel beta, v0.8.2, on Mac OS X 10.13.6 17G2307, locale en-US)
    • Flutter version 0.8.2 at /Users/anatoly/Projects/flutter
    • Framework revision 5ab9e70727 (4 weeks ago), 2018-09-07 12:33:05 -0700
    • Engine revision 58a1894a1c
    • Dart version 2.1.0-dev.3.1.flutter-760a9690c2

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.1)
    • Android SDK at /Users/anatoly/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.0, Build version 10A255
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 3.2 Preview.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] Android Studio (version 3.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 29.0.2
    • Dart plugin version 181.5616
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] IntelliJ IDEA Community Edition (version 2018.1.6)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 26.0.2
    • Dart plugin version 181.4892.1

[!] VS Code (version 1.27.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected devices (1 available)
    • iPhone XS Max • 77F63AD5-6D59-469C-9054-F440BA0AD3CA • ios • iOS 12.0 (simulator)

! Doctor found issues in 1 category.
```

**Comment by pulyaevskiy** [comment]
Ok, so I finally figured it out.

```
targeted OS version does not support use of thread local variables in l1862 for architecture x86_64
```

Looks like this comes from TensorFlow SDK and basically means that you need to set deployment target to iOS 9.0 or higher in XCode (and in your Podspec file).

Versions of iOS before 9.0 do not support thread local variables.

I'll submit a PR to firebase_ml_vision readme explaining this later tonight.

**Comment by rlee1990** [comment]
@pulyaevskiy with the last update yesterday its working just fine now in ios

**Comment by pulyaevskiy** [comment]
@rlee1990 I just tested with the latest version and it was still failing for me. I'm pretty sure this could only be fixed by bumping deployment target to 9.0.

Wondering if you had a different issue by any chance?

**Comment by kroikie** [comment]
@pulyaevskiy

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/622. Any further collaboration will be done there.
